notebooks/byod.py

```diff
@@ -98,27 +98,33 @@ def upload_columns(table: str, columns: Dict[str, List[Any]]):
             tsv_data += os.linesep + "\t".join([f"{i}", *[columns[h][i] for h in column_headers]])
         upload_data_table(tsv_data)
 
+    def parse_cram_crai(filename: str):
+        sample, _ = filename.split(".", 1)                                                                    
+        if filename.endswith(".cram"): 
+            ext = "cram" 
+        elif filename.endswith(".crai") or filename.endswith(".cram.crai"):
+            ext = "crai" 
+        else:
+            # Metadata etc often takes the form of *.txt files so we'll ignore
+            # it rather than raising an exception
+            print(f"Unable to parse {filename} into sample and extension")
+        return sample, ext
+
     def create_cram_crai_table(table: str, listing: Iterable[str]):
         crams = dict()
         crais = dict()
         for key in listing:
             _, filename = key.rsplit("/", 1)
-
-            parts = filename.split(".")
-            if 3 == len(parts):  # foo.cram.crai branch
-                sample, _, ext = parts
-            elif 2 == len(parts):  # "foo.cram" or "foo.crai" branch
-                sample, ext = parts
-            else:
-                raise ValueError(f"Unable to parse '{filename}'")
-
+            ext = parse_cram_crai(filename)
```

Comment:
should be:
```
sample, ext = parse_cram_crai(filename)
```

Reply:
It has been a hot minute since I dealt with functions returning more than one value. Does this return a string of sample and a string of ext, or does it return a list containing two strings? ie does the next section still work?

Reply:
Give it a whirl:
```
In [25]: def foo(): return 23, "blah"                                                                        

In [26]: isinstance(foo(), tuple)                                                                            
Out[26]: True

In [27]: a, b = foo()                                                                                        

In [28]: print(a, b)                                                                                         
23 blah
```